tests/unit/test_event_pipeline/test_mapping_handlers.py

```diff
@@ -0,0 +1,261 @@
+        )
+
+    def test_reset(self):
+        self.handler.reset()
+        self.mock1.reset.assert_called()
+        self.mock2.reset.assert_called()
+        self.mock3.reset.assert_called()
+
+
+class TestKeyHandler(BaseTests, unittest.IsolatedAsyncioTestCase):
+    def setUp(self):
+        self.handler = KeyHandler(
+            EventCombination.from_string("2,0,10+1,3,1"),
+            Mapping(
+                event_combination="2,0,10+1,3,1",
+                target_uinput="mouse",
+                output_symbol="BTN_LEFT",
+            ),
+        )
+
+    def test_reset(self):
+        self.handler.notify(
+            InputEvent(0, 0, EV_REL, REL_X, 1, actions=(EventActions.as_key,)),
+            source=InputDevice("/dev/input/event11"),
+            forward=evdev.UInput(),
+        )
+        history = convert_to_internal_events(
+            global_uinputs.get_uinput("mouse").write_history
+        )
+        self.assertEqual(history[0], InputEvent.from_tuple((EV_KEY, BTN_LEFT, 1)))
+        self.assertEqual(len(history), 1)
+
+        self.handler.reset()
+        history = convert_to_internal_events(
+            global_uinputs.get_uinput("mouse").write_history
+        )
+        self.assertEqual(history[1], InputEvent.from_tuple((EV_KEY, BTN_LEFT, 0)))
+        self.assertEqual(len(history), 2)
+
+
+class TestMacroHandler(BaseTests, unittest.IsolatedAsyncioTestCase):
+    def setUp(self):
+        self.context_mock = MagicMock()
+        self.handler = MacroHandler(
+            EventCombination.from_string("2,0,10+1,3,1"),
+            Mapping(
+                event_combination="2,0,10+1,3,1",
+                target_uinput="mouse",
+                output_symbol="hold_keys(BTN_LEFT, BTN_RIGHT)",
+            ),
+            context=self.context_mock,
+        )
+
+    async def test_reset(self):
+        self.handler.notify(
+            InputEvent(0, 0, EV_REL, REL_X, 1, actions=(EventActions.as_key,)),
+            source=InputDevice("/dev/input/event11"),
+            forward=evdev.UInput(),
+        )
+
+        await asyncio.sleep(0.1)
+        history = convert_to_internal_events(
+            global_uinputs.get_uinput("mouse").write_history
+        )
+        self.assertIn(InputEvent.from_tuple((EV_KEY, BTN_LEFT, 1)), history)
+        self.assertIn(InputEvent.from_tuple((EV_KEY, BTN_RIGHT, 1)), history)
+        self.assertEqual(len(history), 2)
+
+        self.handler.reset()
+        await asyncio.sleep(0.1)
+        history = convert_to_internal_events(
+            global_uinputs.get_uinput("mouse").write_history
+        )
+        self.assertIn(InputEvent.from_tuple((EV_KEY, BTN_LEFT, 0)), history[-2:])
+        self.assertIn(InputEvent.from_tuple((EV_KEY, BTN_RIGHT, 0)), history[-2:])
+        self.assertEqual(len(history), 4)
+
+
+class TestRelToBtnHanlder(BaseTests, unittest.IsolatedAsyncioTestCase):
+    def setUp(self):
+        self.handler = AxisSwitchHandler(
```

Comment:
this creates an AxisSwitchHandler in TestRelToBtnHanlder

Reply:
oh, that looks like an error. I was probably to lazy with the copy paste.